code/socialDistribution/views.py

```diff
@@ -384,21 +397,41 @@ def editPost(request, id):
                 post.unlisted = form.cleaned_data.get('unlisted')
                 post.content_media = content_media
 
-                categories = form.cleaned_data.get('categories').split()
-                previousCategories = Category.objects.filter(post=post)
-                previousCategoriesNames = [
-                    cat.category for cat in previousCategories]
-
-                # Create new categories
-                for category in categories:
-                    if category in previousCategoriesNames:
-                        previousCategoriesNames.remove(category)
-                    else:
-                        Category.objects.create(category=category, post=post)
-
-                # Remove old categories that were deleted
-                for category in previousCategoriesNames:
-                    Category.objects.get(category=category, post=post).delete()
+                categories = form.cleaned_data.get('categories')
+
+                if categories is not None:
+                    categories = categories.split()
+                    categories_to_remove = [ cat.category for cat in post.category_set.all()]
+
+                    """
+                    This implementation makes category names case-insensitive.
+                    This makes handling Category objects cleaner, albeit slightly more
+                    involved.
+                    """
+                    for category in categories:
+                        try:
+                            existing_category = Category.objects.get(category__iexact=category)     # case-insensitive lookup
+                            existing_cat_to_post = existing_category.posts.all().filter(id=post.id)
+                        except Category.DoesNotExist:
+                            existing_category = False
+                        
+                        if not existing_category:
+                            post.category_set.create(category=category)
+
+                        elif not existing_cat_to_post.exists():
+                            existing_category.posts.add(post)
```

Comment:
I find it hard to understand what this code is doing and the purpose of `existing_cat_to_post`. See my other comment about using `post.categories.add` instead as that feels more intuitive. Also, `get_or_create` so that you only ever have to `.add` category and not `.create` category

Reply:
I agree with you! I made some changes based on your comments.

I appreciate your feedback! Thank you! ♥

Let me know if there's anything else that might need some changes.

Reply:
Looks great, thanks! Can you just make sure you merge in master and resolve merge conflicts? You'll also probably need to delete your migration files and rerun `makemigrations`

Reply:
Sounds good 👌 Thanks!